### DISCUSSION TITLE
The TCP connection for the watch was disconnected

### DISCUSSION THREAD

**Discussion by jakub88** [etcd-io/etcd#18769] [Q&A]
My application listens to a key in etcd, but after a period of time (about a week) without any data transmission, the TCP connection was dropped, and the etcd client did not automatically reconnect, causing my application to miss the key change events. After I restarted my application, the watch resumed working normally. How should I troubleshoot this issue?

Below is my client initialization configuration.
`clientV3.Config{
	Endpoints:   etcdEndpoints(),
	DialTimeout: 5 * time.Second,
	TLS:         tlsConf,
},`

**Comment by ahrtr** [comment]
I couldn't reproduce it. I tried to even restart etcd cluster,  the client side watch can still continue to work.
